lib/libutee/tee_api_property.c

```diff
@@ -398,10 +404,42 @@ TEE_Result TEE_GetPropertyAsString(TEE_PropSetHandle propsetOrEnumerator,
 		break;
 
 	case USER_TA_PROP_TYPE_STRING:
-	case USER_TA_PROP_TYPE_BINARY_BLOCK:
 		l = strlcpy(valueBuffer, pv.u.str_val, bufferlen);
 		break;
 
+	/*
+	 * This is returned as a binary blob and thus must be converted
+	 * into base64 per the GP specification.
+	 *
+	 * Note that the behaviour of base64_enc does not copy any data
+	 * into the buffer if all of it will not fit. This appears slightly
+	 * different than the above implementations that copy into the buffer
+	 * as many bytes as will fit and return a short buffer error.
+	 * The GP specification for TEE_GetPropertyAsString does not explicitly
+	 * state this partial data behaviour in response to a short buffer.
+	 */
+	case USER_TA_PROP_TYPE_BINARY_BLOCK: {
+
+		size_t blen = bufferlen;
+
+		if (!base64_enc(pv.u.binary.val, pv.u.binary.len,
+				valueBuffer, &blen)) {
+			assert(blen > bufferlen);
```

Comment:
I would prefer not having these two asserts. We have full control of `base64_enc()` and we should be able to trust it to return correct buffer sizes.